client/recoverSecret.rs

```diff
@@ -0,0 +1,136 @@
+            let mut word = &letters[startInterval..endInterval];
+            // println!("word : {}", word);
+
+            let mut prec_letter:i32;
+            for i in 0..word.len(){
+
+                let mut tmp_index = getValueIndex(result.to_string(), word.chars().nth(i).unwrap().to_string());
+                // If letter of pattern do not exist, add to the end of result
+                if tmp_index==-1 {
+                    result = result.to_owned() + &*word.chars().nth(i).unwrap().to_string();
+                    // println!("result 0: {}", result);
+                }
+                // If letter of pattern do exist
+                else if i!=0 && tmp_index<getValueIndex(result.to_string(), word.chars().nth(i-1).unwrap().to_string()){
+                    // Get precedent letter index of word
+                    prec_letter = getValueIndex(result.to_string(), word.chars().nth(i-1).unwrap().to_string());
+
+                    // Calculate the nex letter index of word
+                    let mut prec_plus = prec_letter + 1;
+
+                    // Get found letter index in result
+                    let mut first_word_letter = getValueIndex(result.to_string(), word.chars().nth(0).unwrap().to_string());
+
+                    // Same in the two conditions but in different case
+                    // - Cut from current index of result to the first letter of word
+                    // - Put it just after the precedent letter index of word
+                    if tmp_index < first_word_letter {
+                        let mut cutted = &result[tmp_index as usize..first_word_letter as usize].to_string();
+                        let new_resutlt = result[..tmp_index as usize].to_string() + &result[first_word_letter as usize..prec_plus as usize].to_string() + &*cutted.to_string() + &result[prec_plus as usize..].to_string();
+                        result = new_resutlt;
+                        // println!("result 1 : {}", result);
+                    }
+                    else{
+                        let mut cutted = &result[tmp_index as usize..prec_letter as usize].to_string();
+                        let new_resutlt = result[0..tmp_index as usize].to_string() + &result[prec_letter as usize..].to_string() + &*cutted.to_string();
+                        result = new_resutlt;
+                        // println!("result 2: {}", result);
+                    }
+                }
+                prec_letter = i as i32;
+            }
+            // println!("\nNew result: {}\n", result);
+        }
+        status = validateRecoverSecretSentence(result.clone().to_string(), tuple_sizes.clone(), letters.clone())
+    }
+
+    let mut output: RecoverSecretOutput = RecoverSecretOutput{
+        secret_sentence: result
+    };
+    return output;
+}
+
+// Get the index of letter parameter in word parameter as parameters
+pub fn getValueIndex(word: String, letter: String) -> i32 {
```

Comment:
put this function at the top because you use it in the function generateRecoverSecretSentence

Reply:
ok